enclave_wrangler/dataset_download.py

```diff
@@ -135,14 +135,53 @@ def combine_parquet_files(input_folder, target_path):
     except Exception as e:
         print(e)
 
-def run(datasetRid: str, ref: str = 'master') -> None:
+
+def cli_validate(args: Dict):
+    """Validate CLI parameters"""
+    # TODO: @Siggie: If we do want `--download-all-registered-datasets` as a param here, I think it would be good to
+    #  throw an error like this. But if you don't want that, or if you think I need to make this message clearer, let me know. - Joe 2022/05/11
+    if args['download_all_registered_datasets'] and (args['datasetName'] or args['datasetRid']):
+        raise RuntimeError('Parameter (1) `--download-all-registered-datasets` was passed, but either/both (2) `--datasetName` and '
+                           '`--datasetRid` were also passed. Please either pass only (1) to download all datasets, or (2) '
+                           'to download 1 dataset, but not both sets of these parameters at the same time.')
+
+
+def run_single(datasetRid: str, ref: str = 'master', write_csv_in_calling_dir=True) -> None:
+    """Run progam"""
     endRef = getTransaction(datasetRid, ref)
     args = {'datasetRid': datasetRid, 'endRef': endRef}
     file_parts = views2(**args)
-    datasets_views(datasetRid, file_parts)
+    df = datasets_views(datasetRid, file_parts)
     # asyncio.run(datasets_views(datasetRid, file_parts))
+    if write_csv_in_calling_dir:
+        df.to_csv()
 
 
+def run_all() -> None:
```

Comment:
Here's what I have so far, quick and dirty. But I stopped, because I got an error.